fennel/internal_lib/to_proto/source_code.py

```diff
@@ -101,15 +100,78 @@ def get_featureset_gen_code(
     return _remove_empty_lines(source_code)
 
 
-def remove_source_and_sink_decorator(text):
-    # Define the regular expression pattern for the @source decorator block
-    pattern = r"^\s*@source\((?:.|\n)*?\)\s*$"
-    # Remove @source decorator using the regex pattern with re.MULTILINE and re.DOTALL flags
-    result = re.sub(pattern, "", text, flags=re.MULTILINE | re.DOTALL)
-    pattern = r"^\s*@sink\((?:.|\n)*?\)\s*$"
-    # Remove @source decorator using the regex pattern with re.MULTILINE and re.DOTALL flags
-    result = re.sub(pattern, "", result, flags=re.MULTILINE | re.DOTALL)
-    return result.strip()
+def get_text_between_ast_pointers(
+    text: str,
+    lineno: int,
+    col_offset: int,
+    end_lineno: int,
+    end_col_offset: int,
+    is_decorator: bool = False,
+):
+    # Split the text into lines
+    lines = text.splitlines()
+
+    if is_decorator:
+        # for adding @
+        col_offset = col_offset - 1
+
+    output = []
+    if lineno == end_lineno:
+        return lines[lineno - 1][col_offset:end_col_offset]
+    for i in range(lineno - 1, end_lineno):
+        if i == lineno - 1:
+            output.append(lines[i][col_offset:])
+        elif i == end_lineno - 1:
+            output.append(lines[i][:end_col_offset])
+        else:
+            output.append(lines[i])
+    return "\n".join(output)
```

Comment:
Can this be simplified - 

    # If the text is within a single line
    if lineno == end_lineno:
        return lines[lineno - 1][col_offset:end_col_offset]
    
    # Collect parts from multiple lines
    parts = [lines[lineno - 1][col_offset:]]  # Start from the initial column offset of the first line
    parts.extend(lines[lineno:end_lineno - 1])  # Add full lines between the first and last line
    parts.append(lines[end_lineno - 1][:end_col_offset])  # End at the final column offset of the last line    
    return "\n".join(parts)

Reply:
changed